packages/bson-bench/src/base.ts

```diff
@@ -28,14 +29,17 @@ function reportErrorAndQuit(error: Error) {
   process.exit(1);
 }
 
-function run(bson: BSONLib, config: BenchmarkSpecification) {
+function run(bson: BSONLib | ConstructibleBSON, config: BenchmarkSpecification) {
   let fn:
     | ((b: Uint8Array, options?: any) => any)
     | ((o: any, options?: any) => Uint8Array)
     | undefined = undefined;
   let documentSizeBytes: number;
   let doc: any;
 
+  // Check if this is bson < v4
+  if (!('serialize' in bson)) bson = new bson();
```

Comment:
nit, don't feel strongly, but "typeof of function" would make sense when the incoming type is either BSON object or BSON constructor